[Deen Liong] iP

deenliong opened the pull request.

deenliong committed:
First commit after accidental deletion.

deenliong committed:
Added test

deenliong committed:
Refactor switch case.

deenliong committed:
Changed array to arraylist and formatted some sentence output

deenliong committed:
Added delete feature.

deenliong committed:
Did some formatting for level 6.

deenliong committed:
Add duke.txt file

sevenseasofbri commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,112 @@
+        System.out.println("What can I do for you?");
+    }
+
+    public static void main(String[] args) {
```
This function is very long (>30lines). Do consider breaking it down into multiple smaller functions.

sevenseasofbri commented on src/main/java/duke/TaskManager.java:
```diff
@@ -0,0 +1,73 @@
+
+public class TaskManager {
+
+    public static void printSuccessfulAdd(Task[] tasks, int taskCount) {
```
Perhaps avoid the usage of magic literals and consider refactoring them to `final` variables instead.

sevenseasofbri commented on src/main/java/duke/TaskManager.java:
```diff
@@ -0,0 +1,73 @@
+
+    public static void printTaskList(Task[] tasks, int taskCount) {
+        System.out.println("\t_____________________");
+        // if there are no task
```
This comment may be redundant. Consider removing it

sevenseasofbri commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,5 @@
+package duke.exception;
+
+public abstract class DukeException extends Exception {
+    //no other code needed
```
You don't need to add comments where it is perhaps not necessary to do so.

sevenseasofbri commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,41 @@
+    }
+
+    public String getStatusIcon() {
+        if (isDone) { // mark done task with X
```
Maybe remove the comment here since it is redundant and the code explains itself well already.

sevenseasofbri commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,41 @@
+    public String toString() {
+
+        return getStatusIcon() + description;
+//        return getStatusIcon() ;
```
Avoid leaving commented out code in your codebase, since it reduces readability

sevenseasofbri reviewed (commented):
Hi Deen, good job on the iP so far! I have added some comments on the code quality, coding standard violations and naming conventions. Hope they will be helpful. 👍🏽

deenliong committed:
Read in file and continue adding to list
small bug - bye now prints error message

deenliong committed:
managed to save new list of items

deenliong committed:
refactor with bugs
1 - bye error message still show
2 - printing of number of items in inputFile at the start

deenliong committed:
solved all bugs

deenliong committed:
slight formatting

deenliong committed:
add meta info

deenliong committed:
simple edit

deenliong committed:
Added UI and custom exception

deenliong committed:
starting on command and parse

deenliong committed:
ui and parser

deenliong committed:
restart moreOOP

deenliong committed:
redo exception

deenliong committed:
todo, mark, list done but loading of files fail

deenliong committed:
starting on deadline and event command

deenliong committed:
done deadlineCommand

deenliong committed:
added all commandparse

deenliong committed:
edited ui

deenliong committed:
amoreOOP without external file

deenliong committed:
Merge pull request #1 from deenliong/branch-Level-9

Branch level 9

deenliong committed:
able to load file

deenliong committed:
added all storage methods

deenliong committed:
Merge branch 'A-MoreOOP'

deenliong committed:
merge moreoop into master

deenliong committed:
added find

deenliong committed:
mark and unmark debugged

deenliong committed:
almost debugged exception with messages

deenliong committed:
level 9 with debugged exceptions

deenliong committed:
debugged getTaskId

deenliong committed:
Merge pull request #2 from deenliong/branch-Level-9

Branch level 9

deenliong committed:
debug

deenliong committed:
format

deenliong committed:
format

deenliong committed:
Added doccumentation for A-JavaDoc

deenliong committed:
added help and debug exception

deenliong committed:
asd

deenliong committed:
nil

deenliong committed:
Added User Guide

deenliong committed:
Merge pull request #3 from deenliong/A-UserGuide

Added User Guide

deenliong committed:
Merge branch 'A-UserGuide' of https://github.com/deenliong/ip

deenliong committed:
Merge branch 'master' of https://github.com/deenliong/ip

deenliong committed:
added total command into list

deenliong committed:
format

deenliong committed:
edited readme

deenliong committed:
edit readme

deenliong committed:
jar done

deenliong committed:
added jar link in readme

deenliong committed:
updated readMe

deenliong committed:
edit readme

deenliong committed:
updated readme
